### PR TITLE
Cookbook

### PR DESCRIPTION
**Pull request by realeinherjar**
- [x] Automated CI in GitHub Actions that test all the code in the cookbook
- [x] Dependabot to update (weekly):
  - [x] GitHub Actions
  - [x] Test dependencies, this is good once we release `rust-bitcoin` new versions dependabot will open a PR here to update
- [ ] TABConf segwit and taproot signing workshop explaining all details

## Details

I want this cookbook to be tested automatically on CI, so the code will always run.
The only solution that worked for me was [this workaround](https://github.com/rust-lang/mdBook/issues/706#issuecomment-1139423009) that is used in [The Rust Rand Book](https://github.com/rust-random/book).

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      # Check for updates to GitHub Actions every week
+      interval: "weekly"
+  - package-ecosystem: "cargo"
+    directory: "cookbook/tests/"
+    schedule:
+      # Check for updates to Rust dependencies every week
+      interval: "weekly"
```
Comment (realeinherjar) [inline .github/dependabot.yml]: NOTE to myself: I need to test this (first time using the `package-ecosystem: "cargo"`).

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          mdbook-version: 'latest'
+
+      - run: mdbook build
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./cookbook/book
```
Comment (realeinherjar) [inline .github/workflows/gh-pages.yml]: @tcharding this is totally wrong, I am adding as a placeholder now.
I'll update once I finish to add the content.
Feel free to comment, edit, contribute...

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master, '0.[0-9]+' ]
+  pull_request:
+    branches: [ master, '0.[0-9]+' ]
```
Comment (realeinherjar) [inline .github/workflows/test.yml]: NOTE: we might use `v0.[0-9]+`

File: cookbook/book.toml [hunk 4]
```diff
@@ -1,6 +1,10 @@
 [book]
-authors = ["Harshil Jani"]
+authors = ["Toby Harding<me@tobin.cc>", "Einherjar <realeinherjar@proton.me>"]
```
Comment (realeinherjar) [inline cookbook/book.toml]: @tcharding I have no idea what to put in here...

Comment (tcharding) [reply cookbook/book.toml]: Leave Harshil's name in there please because he did some prior work. Definitely throw your name in there since you are working on it. Leave mine out for now please because I have not added anything meaningful, I'll add it later if/when I do. (I'm Tobin BTW :)

Comment (realeinherjar) [reply cookbook/book.toml]: Sorry I thought Harshil was a placeholder 😓

**Comment by realeinherjar** [comment]
I screwup the rebase. Let me try again :/
